CAM-12052: document telemetry (#713)
camunda/camunda-docs-manual: content/update/minor/713-to-714/_index.md

1. For administrators and developers: [Database Updates](#database-updates)
1. For administrators and developers: [Full Distribution Update](#full-distribution)
1. For administrators: [Standalone Web Application](#standalone-web-application)
1. For developers: [Update to JQuery 3.5](#update-to-jquery-3-5)
1. For developers: [Changes to Task Query and Historic Task Query behavior](#changes-to-task-query-and-historic-task-query-behavior)
1. For developers: [New Engine Dependency - Connect](#new-engine-dependency-connect)
1. For developers: [Cockpit Style Customizations](#cockpit-style-customizations)
1. For developers: [Changes to the Cockpit Config File](#cockpit-config-file)
1. For developers: [New Frontend Plugin System for Cockpit](#cockpit-plugins)
1. For developers: [End of Spring 3 Support](#end-of-spring-3-support)
1. For developers: [New Process Engine Property](#new-process-engine-property)
1. For developers: [Disable Telemetry Reporter in Tests](#disable-telemetry-reporter-in-tests)

This guide covers mandatory migration steps as well as optional considerations for the initial configuration of new functionality included in Camunda BPM 7.14.

# Legal Note

> Before you upgrade a Camunda BPM Runtime version >= 7.14.0-alpha1 or activate the telemetry functionality, please make sure that you are authorized to take this step, and that the installation or activation of the [telemetry functionality][engine-config-initializeTelemetry] is not in conflict with any company-internal policies, compliance guidelines, any contractual or other provisions or obligations of your company.

Reviewer: ```suggestion
Before you upgrade to a Camunda BPM Runtime version >= 7.14.0-alpha1 or activate the telemetry functionality, please make sure that you are authorized to take this step, and that the installation or activation of the [telemetry functionality][engine-config-initializeTelemetry] is not in conflict with any company-internal policies, compliance guidelines, any contractual or other provisions or obligations of your company.
```

Camunda cannot be held responsible in the event of unauthorized installation or activation of this function.

> Read further about the telemetry topic, [here]({{< ref "/introduction/telemetry.md" >}}).

Reviewer: ```suggestion
You can find more details on the telemetry topic in the [general introduction][telemetry].
```

# New Process Engine Property

This minor release introduces a new process engine property - `camunda.installation.id`. In case you already created a property with the same name, please consider renaming it, to prevent collisions with the process engine code. So far the property will be used for the [telemetry][] feature, however, in the future that might change and cause issues with implemented features. To perform the renaming of an existing property, please delete first the property, store your property with different name and restart the engine. The new `camunda.installation.id` property will be created during the start of the process engine and stored into the database. Here is an example how the change can be performed:

```java
managementService.deleteProperty("camunda.installation.id");
managementService.setProperty(customCamundaInstallatioId, customCamundaInstallatioIdValue);
// keep in mind to restart the engine afterward
```

In case the custom installation id property is not adjusted, the telemetry feature most probably will be not functioning correctly.

# Disable Telemetry Reporter in Tests

> With the new [telemetry][] feature and on a strict opt-in basis, we are looking to collect environment and usage data to further improve the user experience for you. However, there are certain scenarios that it will be better to disable completely the telemetry functionality. One of those scenarios is in tests that included process engine usage. Therefore we highly recommend you to disable the telemetry reporter to prevent sending any requests from testing:

Reviewer: ```suggestion
With the new [telemetry][] feature and on a strict opt-in basis, we are looking to collect environment and usage data to further improve the user experience for you. However, for certain scenarios it will be better to completely disable the telemetry functionality. One of those scenarios is testing, if that includes process engine usage. Therefore we highly recommend to disable the telemetry reporter to prevent sending any requests from your tests:
```

> * via xml based configuration file: [example]({{<ref "/user-guide/testing/_index.md#disabling-telemetry">}})

Reviewer: ```suggestion
* via XML-based configuration file: [example]({{<ref "/user-guide/testing/_index.md#disabling-telemetry">}})
```

> * via xml based configuration file: [example]({{<ref "/user-guide/spring-boot-integration/testing.md#disabling-telemetry">}})

Reviewer: ```suggestion
* via YAML-based configuration file in Spring Boot: [example]({{<ref "/user-guide/spring-boot-integration/testing.md#disabling-telemetry">}})
```
